analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.TryCatchFinally.cs

```diff
@@ -464,11 +464,8 @@ public void Finally_NestedFinally()
             ValidateHasOnlyNoExceptionAndUnknownException(validator, "InInnerFinally");
             ValidateHasOnlyNoExceptionAndUnknownException(validator, "InInnerTry");
 
-            validator.TagStates("AfterInnerFinally").Should().HaveCount(1)
-                     .And.ContainSingle(x => HasNoException(x));
-
-            validator.TagStates("AfterOuterFinally").Should().HaveCount(1)
-                     .And.ContainSingle(x => HasNoException(x));
+            validator.TagStates("AfterInnerFinally").Single().Should().HaveNoException();
```

Comment:
If we would have 2 or more states:
_Before:_ The assertion would fail with `` - pretty clear, straightforward, easy to read in UI.
```
  Message: 
    Expected validator.TagStates("InCatch") to contain 1 item(s), but found 2.
```
_After:_ The nice message is lost in case the count changes.
```
  Message: 
    Test method SonarAnalyzer.UnitTest.SymbolicExecution.Roslyn.RoslynSymbolicExecutionTest.TryCatchFinally_ThrowInTry threw exception: 
    System.InvalidOperationException: Sequence contains more than one element

  Stack Trace: 
    ThrowHelper.ThrowMoreThanOneElementException()
    Enumerable.TryGetSingle[TSource](IEnumerable`1 source, Boolean& found)
    Enumerable.Single[TSource](IEnumerable`1 source)
    RoslynSymbolicExecutionTest.TryCatchFinally_ThrowInTry() line 655
```
All of the usages are on `ProgramState[]`, can we update the assertion framework to assert on the array or `IEnumerable<ProgramState>`? And assert the count inside?

We don't have the need for the single ProgramState assertions (yet), so I'd update the existing classes instead of adding another layer.

Reply:
I've tried but it's not that easy to do the registration with Should.

I want to start with baby steps. Do this for now and add later the checks for the array.